go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -914,7 +914,7 @@ func (be *BuiltinBackupEngine) restoreFiles(ctx context.Context, params RestoreP
 			select {
 			case <-ctx.Done():
 				log.Errorf("Context cancelled during %q backup", fe.Name)
-				bh.RecordError(vterrors.Errorf(vtrpc.Code_CANCELED, "context canceled"))
+				rec.RecordError(vterrors.Errorf(vtrpc.Code_CANCELED, "context canceled"))
```

Comment:
bug left in https://github.com/vitessio/vitess/pull/12703

Reply:
We didn't have restore test so we weren't able to catch it .. now that I wrote restore test but it will not catch this until we fix https://github.com/vitessio/vitess/issues/12830